node-util/sbin/oo-idler-stats

```diff
@@ -144,40 +143,54 @@ def get_app_type(app_path):
     except:
         return "unknown"
 
+app_name = ""
 
 def validate(uuid):
     """ given a user's uuid, verify an app exists. """
     app_path = "/var/lib/openshift/%s" % uuid
 
-    # Is there a /git directory?
-    if not os.path.isdir(app_path + "/git"):
-        return "Path %s/git not found." % app_path
-
-    # Is there something inside the /git directory?
-    dirs = os.listdir(app_path + "/git")
-    if len(dirs) == 0:
-        return "Path %s/git is empty." % app_path
-
-    # Under /git should be appname.git
-    app_name = None
-    for d in dirs:
-        if re.search('\.git$', d):
-            app_name = d.replace(".git", "")
-
-    # Is there stuff inside git/appname.git?
-    gitfiles = os.listdir(app_path + "/git/" + app_name + ".git")
-    if len(gitfiles) == 0:
-        return "Path %s/git/%s.git is empty." % (app_path, app_name)
-
-    # Chances are good that this is a git repo. What's the last commit?
-    last_commit = run('git show HEAD | grep Date | sed "s/Date:   //"',
-                      app_path + "/git/" + app_name + ".git")[1][0]
+    global app_name
```

Comment:
This means that you are assuming that `validate` will be called first on a gear with a framework cartridge, and then subsequently on any other gears from the same application, correct? Looking at https://github.com/dinhxuanvu/origin-server/blob/idler-stats/node-util/sbin/oo-idler-stats#L231-L247, it is not obvious to me (1) that gears will be grouped by application and (2) that gears will be sorted such that a gear with a framework cartridge comes before related gears without.